app/Models/BeatmapsetArchive.php

```diff
@@ -5,13 +5,18 @@
 
 namespace App\Models;
 
+use App\Exceptions\BeatmapProcessorException;
+
 class BeatmapsetArchive
 {
     public function __construct(string $osz)
     {
         $this->osz = $osz;
         $this->zip = new \ZipArchive();
-        $this->zip->open($this->osz);
+        $code = $this->zip->open($this->osz);
+        if ($code !== true) {
```

Comment:
`__destruct`'s `close` call can use a check if it's open before trying to close otherwise it'll throw another error (at least in console).

```
>>> $b = new App\Models\BeatmapsetArchive('/tmp/junk')
App\Exceptions\BeatmapProcessorException with message 'Failed to open archive'
>>>

Exit:  Ctrl+D
PHP Fatal error:  Uncaught ValueError: Invalid or uninitialized Zip object in /home/edho/osu/osu-web/app/Models/BeatmapsetArchive.php:24
Stack trace:
#0 /home/edho/osu/osu-web/app/Models/BeatmapsetArchive.php(24): ZipArchive->close()
#1 [internal function]: App\Models\BeatmapsetArchive->__destruct()
#2 {main}
  thrown in /home/edho/osu/osu-web/app/Models/BeatmapsetArchive.php on line 24
```